File: optics_9t9it/doc_events/customer.py
# -*- coding: utf-8 -*-
# Copyright (c) 2019, 9T9IT and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

from optics_9t9it.api.customer import get_user_branch


def before_naming(doc, method):
    if not doc.branch:
        doc.branch = get_user_branch()

File: 9t9it_optics/public/js/scripts/customer.js
async function set_branch(frm) {
  if (frm.doc.__islocal) {
    const { message: branch } = await frappe.call({
      method: '9t9it_optics.api.customer.get_user_branch',
    });
    console.log(branch);
    frm.set_value('branch', branch);
  }
}

export default {
  onload: set_branch,
};
